src/pyvesync/vesync.py

```diff
@@ -395,12 +395,18 @@ def energy_today(self):
     @abstractmethod
     def power(self):
         """Return current power in watts"""
-        return float(self.details.get('power'))
+        try:
+            return float(self.details.get('power'))
+        except (TypeError, ValueError):
+            return 0
 
     @abstractmethod
     def voltage(self):
         """Return current voltage"""
-        return float(self.details.get('voltage'))
+        try:
```

Comment:
what do you think about ditching the try/catch and using this instead to keep it simple?

`return float(self.details.get('voltage', 0))`

That should deal with the case where the dict does not contain the key by setting the default value to 0 which can be cast to float

Reply:
That's much more elegant